--- libstemmer_c-2.2.0/Makefile.orig	2024-09-14 12:50:10.978761983 +0200
+++ libstemmer_c-2.2.0/Makefile	2024-09-14 12:55:31.260360202 +0200
@@ -2,12 +2,47 @@ include mkinc.mak
 ifeq ($(OS),Windows_NT)
 EXEEXT=.exe
 endif
-CFLAGS=-O2
+CFLAGS?=-O2
+CFLAGS+=-D_REENTRANT
 CPPFLAGS=-Iinclude
-all: libstemmer.a stemwords$(EXEEXT)
-libstemmer.a: $(snowball_sources:.c=.o)
-	$(AR) -cru $@ $^
-stemwords$(EXEEXT): examples/stemwords.o libstemmer.a
-	$(CC) $(CFLAGS) -o $@ $^
+
+bindir=/usr/bin
+libdir=/usr/lib
+includedir=/usr/include
+DESTDIR=
+
+LIBTOOL=libtool
+INSTALL=install -c
+
+LT_CURRENT=1
+LT_REVISION=2
+LT_AGE=1
+
+LTOBJS:= $(snowball_sources:.c=.lo)
+
+all: libstemmer.la stemwords$(EXEEXT)
+
+libstemmer.la: $(LTOBJS)
+	$(LIBTOOL) --tag=CC --mode=link $(CC) $(CFLAGS) $(CPPFLAGS) -rpath $(libdir) \
+	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) -o \
+	$@ $(LTOBJS)
+
+stemwords$(EXEEXT): examples/stemwords.o libstemmer.la
+	$(LIBTOOL) --tag=CC --mode=link $(CC) $(CFLAGS) -o $@ $(LDFLAGS) $^
+
+%.lo: %.c
+	$(LIBTOOL) --tag=CC --mode=compile $(CC) -c $(CFLAGS) $(CPPFLAGS) $< -o $@
+
+install:
+	$(INSTALL) -d $(DESTDIR)$(bindir)
+	$(INSTALL) -d $(DESTDIR)$(libdir)
+	$(INSTALL) -d $(DESTDIR)$(includedir)/libstemmer
+	$(LIBTOOL) --mode=install $(INSTALL) libstemmer.la $(DESTDIR)$(libdir)
+	$(LIBTOOL) --mode=install $(INSTALL) -m 755 stemwords $(DESTDIR)$(bindir)
+	$(INSTALL) -m 644 runtime/api.h $(DESTDIR)$(includedir)/libstemmer
+	$(INSTALL) -m 644 runtime/header.h $(DESTDIR)$(includedir)/libstemmer
+	$(INSTALL) -m 644 include/libstemmer.h $(DESTDIR)$(includedir)/libstemmer
+
 clean:
-	rm -f stemwords$(EXEEXT) libstemmer.a *.o src_c/*.o examples/*.o runtime/*.o libstemmer/*.o
+	rm -f stemwords$(EXEEXT) libstemmer.la *.o *.lo src_c/*.o examples/*.o runtime/*.o libstemmer/*.o
+	rm -rf .libs
